### DISCUSSION TITLE
Why `convert` do so ugly PNG from SVG?

### DISCUSSION THREAD

**Discussion by igormironchik** [ImageMagick/ImageMagick#6586] [Help] [answered]
### ImageMagick version

7.1.1-16

### Operating system, version and so on

Kubuntu 22.04

### Description

Why ImageMagick's `convert` utility do so ugly PNG from the following SVG? Who is wrong here? Maybe it's an issue of some dependency library?

- SVG
  ![1](https://github.com/ImageMagick/ImageMagick/assets/11455533/a8cd8bc7-4b2c-45ea-9a12-669f73fce796)

- PNG
  ![1](https://github.com/ImageMagick/ImageMagick/assets/11455533/82f8b48a-ab94-452f-8776-d6336d8cbe1f)

**Comment by fmw42** [comment]
What was your command?  What SVG renderer are you using (Inkscape, RSVG or IM's MSVG/XML)?

Try supersampling:

`convert -density 288 image.svg -resize 25% image.png`

If that does not help, install Inkscape and try it again.

**Comment by igormironchik** [comment]
@fmw42

I used `convert 1.svg 1.png`

Your command made the following PNG - ![2](https://github.com/ImageMagick/ImageMagick/assets/11455533/85f87ad8-5f80-43b7-be97-03ec4790cc80)

In Inkscape 1.1.2 I see the same issue, whereas Firefox, Microsoft Edge renders such SVG fine (you can see how browser renders this SVG in the question - there is everything fine).

**Comment by fmw42** [comment]
If I do (IM 6 with Inkscape)

`convert -density 288 image.svg x.png`

I get

![x](https://github.com/ImageMagick/ImageMagick/assets/13458218/ea673ed7-b078-40d2-a094-d3465cb62539)

What is wrong with that?  Is it the black on the right side?

This is an issue with Inkscape or possibly your SVG file.  You should report it to Inkscape unless some other IM user can see an issue in your file?    What version of Inkscape are you using?  I am using v1.2.2

**Comment by igormironchik** [comment]
> What is wrong with that? Is it the black on the right side?

Yes, exactly, I talk about black on right side.

> What version of Inkscape are you using?

1.1.2

How Inkscape correlates with IM?

**Comment by snibgo** [comment]
If I run Inkscape v1.2 on stars.svg, it shows an image like the one in fmw42's post.

If that isn't what is wanted, then change the SVG. If it is caused by an Inkscape problem, then report it.

**Comment by igormironchik** [comment]
I reported the problem to Inkscape.

https://gitlab.com/inkscape/inbox/-/issues/9125

But how does Inkscape correlate with ImageMagick?

**Reply by fmw42** [reply, marked as answer]
Imagemagick uses other tools to read SVG files -- Inkscape, RSVG or IM's internal MSVG/XML renderers. If Inkscape is on your system where IM can find it, then that will be used in place of either of the other two.
